app/src/main/java/com/botsone/android/amoleddit/ArticleActivity.java

```diff
@@ -91,12 +106,56 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                 String permalink = currentArticle.getPermalink();
 
                 // Create a new intent to send user to detail activity
-                Intent detailIntent = new Intent(ArticleActivity.this, DetailActivity.class);
+                final Intent detailIntent = new Intent(ArticleActivity.this, DetailActivity.class);
+
+                ImagePipeline imagePipeline = Fresco.getImagePipeline();
+                Uri uri = Uri.parse(currentArticle.getImageUrl());
+
+                ImageRequest request = ImageRequestBuilder
+                        .newBuilderWithSource(uri).build();
+
+                imagePipeline.prefetchToBitmapCache(request, this);
+
+                DataSource<CloseableReference<CloseableImage>>
+                        dataSource = imagePipeline.fetchDecodedImage(request, this);
+
+                dataSource.subscribe(
+                        new BaseBitmapDataSubscriber() {
+                            @Override
+                            public void onNewResultImpl(@Nullable Bitmap bitmap) {
+                                // Pass bitmap to system, which makes a copy of the bitmap.
+                                File cacheDir = getBaseContext().getCacheDir();
+                                File f = new File(cacheDir, "pic");
+
+                                try {
+                                    FileOutputStream out = new FileOutputStream(
+                                            f);
+                                    bitmap.compress(
+                                            Bitmap.CompressFormat.JPEG,
+                                            100, out);
+                                    out.flush();
+                                    out.close();
+
+                                } catch (FileNotFoundException e) {
+                                    e.printStackTrace();
+                                } catch (IOException e) {
+                                    e.printStackTrace();
+                                }
+                                // No need to do any cleanup.
+                            }
+
+                            @Override
+                            public void onFailureImpl(DataSource dataSource) {
+                                // No cleanup required here.
+                            }
+                        }, CallerThreadExecutor.getInstance());
+
 
                 detailIntent.putExtra("key", parsedUri);
```

Comment:
Not a change you made just some feedback
1. Let's move strings like this to constants: Define these at the top of the file as constants:

```
String PARSED_URI_KEY = "parsedUri";
String TITLE_KEY = "title";
String PERMALINK_KEY = "permalink";
```
Then use them: `detailIntent.putExtra(PARSED_URI_KEY, parsedUri);`

2. Make "key" more descriptive.



Same thing in places like here:

`String url = "http://www.reddit.com/r/amoledbackgrounds";`

Pull that out to a constant at the top of the file, it never changes.

Readable string (stuff that the user can read) like below should be pulled out into `strings.xml` so they are easier to localize. Other strings like uri's and shit like that that people don't read, should be pulled to constants.

`String shareMessage= "\nCheck out this wallpaper app called Amoleddit!\n\n";`